UnionPatcher.Gui/MainForm.cs

```diff
@@ -42,32 +43,49 @@ public class MainForm : Form {
             button.Click += delegate {
                 dialog.Close();
             };
-            
+
             return dialog;
         }
-        
+
         public Control CreatePatchButton(int tabIndex = 0) {
             Button control = new() {
                 Text = EasterEgg.Restitch ? "Restitch!" : "Patch!",
                 TabIndex = tabIndex,
             };
 
             control.Click += delegate {
-                if(string.IsNullOrEmpty(this.filePicker.FilePath)) {
+                if(string.IsNullOrWhiteSpace(this.filePicker.FilePath)) {
                     this.CreateOkDialog("Form Error", "No file specified!").ShowModal();
                     return;
                 }
 
-                if(string.IsNullOrEmpty(this.serverUrl.Text)) {
+                if(string.IsNullOrWhiteSpace(this.serverUrl.Text)) {
                     this.CreateOkDialog("Form Error", "No server URL specified!").ShowModal();
                     return;
                 }
 
-                if(string.IsNullOrEmpty(this.outputFileName.FilePath)) {
+                if(string.IsNullOrWhiteSpace(this.outputFileName.FilePath)) {
                     this.CreateOkDialog("Form Error", "No output file specified!").ShowModal();
                     return;
                 }
 
+                ElfFile eboot = new(this.filePicker.FilePath);
+
+                if(eboot.IsValid == false) {
+                    this.CreateOkDialog("Eboot Error", $"{eboot.Name} is not a valid ELF file (magic number mismatch)").ShowModal();
```

Comment:
Hmm, I think it would be a bit better if we explained the problem a little more.

Perhaps something like this?
```suggestion
                    this.CreateOkDialog("Eboot Error", $"{eboot.Name} is not a valid ELF file (magic number mismatch)\nYou must decrypt the EBOOT.bin first.").ShowModal();
```

Reply:
Added a slightly more helpful message